### ISSUE TITLE
Duplicate variables in switch statements

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#125]
Is the following code legal? Should quick-lint-js report an issue for this code?

```js
switch (x) {
  case 1:
    let y;
    break;
  case 2:
    let y;
    break;
}
```

**Comment by strager** [comment]
quick-lint-js does the right thing for this code, IMO:
```
hello.js:6:9: error: redeclaration of variable: y [E034]
hello.js:3:9: note: variable already declared here [E034]
hello.js:1:9: warning: use of undeclared variable: x [E057]
```
